src/sbml/test/TestSBMLErrorLog.c

```diff
@@ -0,0 +1,168 @@
+#include <sbml/xml/XMLError.h>
+#include <sbml/xml/XMLNamespaces.h>
+#include <sbml/SBMLDocument.h>
+
+#include <check.h>
+
+
+
+#include <sbml/common/extern.h>
+
+LIBSBML_CPP_NAMESPACE_USE
+
+
+BEGIN_C_DECLS
+
+START_TEST(test_SBMLErrorLog_removeAll)
+{
+  SBMLErrorLog_t *log = SBMLErrorLog_create();
+  XMLError_t *e = XMLError_createWithIdAndMessage(1234, "1");
+
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e));
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e));
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e));
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e));
+  
+  fail_unless(SBMLErrorLog_contains(log, 1234) == true);
+  fail_unless(SBMLErrorLog_getNumErrors(log) == 4);
+  
+  SBMLErrorLog_remove(log, 1234);
+  fail_unless(SBMLErrorLog_contains(log, 1234) == true);
+  fail_unless(SBMLErrorLog_getNumErrors(log) == 3);
+
+  SBMLErrorLog_removeAll(log, 1234);
+  fail_unless(SBMLErrorLog_contains(log, 1234) == false);
+  fail_unless(SBMLErrorLog_getNumErrors(log) == 0);
+
+  XMLError_free(e);
+  SBMLErrorLog_free(log);
+
+}
+END_TEST
+
+
+START_TEST(test_SBMLErrorLog_add_clear)
+{
+  SBMLErrorLog_t *log = SBMLErrorLog_create();
+  XMLError_t *e = XMLError_createWithIdAndMessage(1234, "1");
+  XMLError_t *e1 = XMLError_createWithIdAndMessage(2345, "1");
+
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e));
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e1));
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e));
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e));
+
+  fail_unless(SBMLErrorLog_contains(log, 1234) == true);
+  fail_unless(SBMLErrorLog_contains(log, 2345) == true);
+  fail_unless(SBMLErrorLog_getNumErrors(log) == 4);
+
+  SBMLErrorLog_remove(log, 2345);
+  fail_unless(SBMLErrorLog_contains(log, 1234) == true);
+  fail_unless(SBMLErrorLog_contains(log, 2345) == false);
+  fail_unless(SBMLErrorLog_getNumErrors(log) == 3);
+
+  SBMLErrorLog_clearLog(log);
+  fail_unless(SBMLErrorLog_getNumErrors(log) == 0);
+
+  XMLError_free(e);
+  XMLError_free(e1);
+  SBMLErrorLog_free(log);
+}
+END_TEST
+
+
+START_TEST(test_SBMLErrorLog_get)
+{
+  SBMLErrorLog_t *log = SBMLErrorLog_create();
+  XMLError_t *e = XMLError_createWithIdAndMessage(1234, "1");
+  XMLError_t *e1 = XMLError_createWithIdAndMessage(2345, "1");
+
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e));
+  SBMLErrorLog_add(log, (const SBMLError_t*)(e1));
+
+  const SBMLError_t* r = SBMLErrorLog_getError(log, 1);
+
+  fail_unless(r != NULL);
+  fail_unless(XMLError_getSeverity(r) == 1);
+  fail_unless(XMLError_getErrorId(r) == 2345);
+
+  const SBMLError_t* r1 = SBMLErrorLog_getErrorWithSeverity(log, 0, 1);
+
+  fail_unless(r1 != NULL);
+  fail_unless(XMLError_getSeverity(r1) == 1);
+  fail_unless(XMLError_getErrorId(r1) == 1234);
+
+  fail_unless(SBMLErrorLog_getNumFailsWithSeverity(log, 1) == 2);
+
+  XMLError_free(e);
+  XMLError_free(e1);
+  //XMLError_free((SBMLError_t*)(r));
+  //XMLError_free((SBMLError_t*)(r1));
+  SBMLErrorLog_free(log);
+}
+END_TEST
+
```

Comment:
I think we need tests that test retrieving SBMLErrors when they are XMLErrors ensuring we don't garble ptrs.